tests/unit/storages/test_request_queue.py

```diff
@@ -162,3 +163,49 @@ async def test_add_batched_requests(
 
     # Confirm the queue is empty after processing all requests
     assert await request_queue.is_empty() is True
+
+
+async def test_invalid_user_data_serialization() -> None:
+    with pytest.raises(ValidationError):
+        Request.from_url(
+            'https://crawlee.dev',
+            user_data={
+                'foo': datetime(year=2020, month=7, day=4, tzinfo=timezone.utc),
+                'bar': {datetime(year=2020, month=4, day=7, tzinfo=timezone.utc)},
+            },
+        )
```

Comment:
Not sure where to put this one, but it is similar to the ones that follow...

Reply:
I believe we can move it to a separate file `test_request.py`.